### PR TITLE
Feat/water 2861 notify callbacks

### PR DESCRIPTION
**Pull request by timtamimi**
Original ticket: https://eaflood.atlassian.net/browse/WATER-2861

### CODE REVIEW

File: test/external/modules/notify/controller.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  });
+
+  experiment('when it has no token', () => {
+    before(async () => {
+      request.headers = {};
+    });
+    test('returns Unauthorized', async () => {
+      await controller.callback(request, h);
+      expect(h.response.calledWith('Unauthorized')).to.be.true();
+      expect(h.response.lastCall.returnValue.code.lastCall.args).to.equal([403]);
```
Comment (DGajewska) [inline test/external/modules/notify/controller.js:39]: This is fine as is, but setting up your stubs slightly differently, could make this a lot easier to read:
```
const responseStub = {
      code: sandbox.stub()
    };

h = {
    response: sinon.stub().returns(responseStub),
   ... }
```

Your test can then be writted as `expect(responseStub.code.calledWith(403)).to.be.true();`

File: test/regression/external/notify-callback.spec.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const { sleepFor } = require('../shared/helpers/utils');
+const config = require('./config');
+
+/* eslint-disable no-undef */
+describe('notify callback attempt', function () {
+  let scheduledNotification, scheduledNotificationAfterCallback, scheduledNotificationNotifyId;
+  before(async () => {
+    await sleepFor(1000);
+    scheduledNotification = await getLastNotifications(config.baseUrl, config.userEmails.notifyCallbackTestEmail);
+    console.log(scheduledNotification);
```
Comment (DGajewska) [inline test/regression/external/notify-callback.spec.js:11]: Did you mean to leave this in?

File: test/regression/external/notify-callback.spec.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  it('checks if the notification was found', () => {
+    expect(scheduledNotificationNotifyId.length).toBe(36);
+  });
+
+  it('calls the notify callback endpoint', async () => {
+    await simulateNotifyCallback(scheduledNotificationNotifyId);
+    await sleepFor(1000);
+    scheduledNotificationAfterCallback = await getLastNotifications(config.baseUrl, config.userEmails.notifyCallbackTestEmail);
+    expect(JSON.parse(scheduledNotificationAfterCallback).data[0].notify_status).toBe('delivered');
```
Comment (DGajewska) [inline test/regression/external/notify-callback.spec.js:23]: Would it be cleaner to abstract away parsing the JSON data and returning `data` from the `getLastNotifications` helper function?

**Review by DGajewska — APPROVED** [review]

File: test/regression/external/notify-callback.spec.js (hunk 2 above)
Comment (timtamimi) [reply test/regression/external/notify-callback.spec.js:11]: 🙊 oops. No, I will remove this. Thank you.

File: test/regression/external/notify-callback.spec.js (hunk 3 above)
Comment (timtamimi) [reply test/regression/external/notify-callback.spec.js:23]: Yep. That makes sense. I will do that.

File: test/external/modules/notify/controller.js (hunk 1 above)
Comment (timtamimi) [reply test/external/modules/notify/controller.js:39]: Cool. That seems tidier so I'll go with that :)

**Pull request merged**